lib/ocpp/common/websocket/websocket_libwebsockets.cpp

```diff
@@ -1195,7 +1204,26 @@ int WebsocketTlsTPM::process_callback(void* wsi_ptr, int callback_reason, void*
     } break;
 
     case LWS_CALLBACK_CLIENT_RECEIVE:
-        on_message(in, len);
+        // We have a buffered message on the pipe
+        if (recv_buffered_message.size() > 0) {
+            recv_buffered_message.append(reinterpret_cast<char*>(in), reinterpret_cast<char*>(in) + len);
+
+            if (lws_remaining_packet_payload(wsi) <= 0) {
+                // Last piece, dispatch
+                on_message(std::move(recv_buffered_message));
+                recv_buffered_message.clear();
+            }
+        } else {
+            if (lws_remaining_packet_payload(wsi) > 0) {
+                // Append to internal message
+                recv_buffered_message.append(reinterpret_cast<char*>(in), reinterpret_cast<char*>(in) + len);
+            } else {
+                // Simply dispatch
+                std::string simple_message(reinterpret_cast<char*>(in), reinterpret_cast<char*>(in) + len);
+                on_message(std::move(simple_message));
+            }
+        }
+
```

Comment:
This could simply be:

```suggestion
    recv_buffered_message.append(reinterpret_cast<char*>(in), reinterpret_cast<char*>(in) + len);
        
        // Message is complete
        if (lws_remaining_packet_payload(wsi) <= 0) {
            on_message(std::move(recv_buffered_message));
            recv_buffered_message.clear();
        }
```

Reply:
Makes sense, the previous logic was a bit unnecessarily complicated.